components/connectivity-adapter/internal/appregistry/service/converter.go

```diff
@@ -12,25 +21,420 @@ func NewConverter() *converter {
+			if outApi.DefaultAuth == nil {
+				outApi.DefaultAuth = &graphql.AuthInput{}
+			}
+			if deprecated.Api.RequestParameters.Headers != nil {
+				h := (graphql.HttpHeaders)(*deprecated.Api.RequestParameters.Headers)
+				outApi.DefaultAuth.AdditionalHeaders = &h
+			}
+			if deprecated.Api.RequestParameters.QueryParameters != nil {
+				q := (graphql.QueryParams)(*deprecated.Api.RequestParameters.QueryParameters)
+				outApi.DefaultAuth.AdditionalQueryParams = &q
+			}
+		}
+
+		if deprecated.Api.Spec != nil {
+			if outApi.Spec == nil {
+				outApi.Spec = &graphql.APISpecInput{}
+			}
+			c := graphql.CLOB(string(deprecated.Api.Spec))
+			outApi.Spec.Data = &c
+			if outApi.Spec.Type == "" {
+				outApi.Spec.Type = graphql.APISpecTypeOpenAPI
+			}
+			if outApi.Spec.Format == "" {
+				outApi.Spec.Format = graphql.SpecFormatJSON
+			}
+		}
+
+		if deprecated.Api.SpecificationUrl != "" {
+			if outApi.Spec == nil {
+				outApi.Spec = &graphql.APISpecInput{}
+			}
+			outApi.Spec.FetchRequest = &graphql.FetchRequestInput{
+				URL: deprecated.Api.SpecificationUrl,
+			}
+
+			if deprecated.Api.SpecificationCredentials != nil || deprecated.Api.SpecificationRequestParameters != nil {
+				outApi.Spec.FetchRequest.Auth = &graphql.AuthInput{}
+			}
+
+			if deprecated.Api.SpecificationCredentials != nil {
+				if deprecated.Api.SpecificationCredentials.Oauth != nil {
+					inOauth := deprecated.Api.SpecificationCredentials.Oauth
+					outApi.Spec.FetchRequest.Auth.Credential = &graphql.CredentialDataInput{
+						Oauth: &graphql.OAuthCredentialDataInput{
+							URL:          inOauth.URL,
+							ClientID:     inOauth.ClientID,
+							ClientSecret: inOauth.ClientSecret,
+						},
+					}
+				}
+				if deprecated.Api.SpecificationCredentials.Basic != nil {
+					inBasic := deprecated.Api.SpecificationCredentials.Basic
+					outApi.Spec.FetchRequest.Auth.Credential = &graphql.CredentialDataInput{
+						Basic: &graphql.BasicCredentialDataInput{
+							Username: inBasic.Username,
+							Password: inBasic.Password,
+						},
+					}
+				}
+			}
+		}
+
+		if deprecated.Api.SpecificationRequestParameters != nil {
+			if deprecated.Api.SpecificationRequestParameters.Headers != nil {
+				h := (graphql.HttpHeaders)(*deprecated.Api.SpecificationRequestParameters.Headers)
+				outApi.Spec.FetchRequest.Auth.AdditionalHeaders = &h
+			}
+			if deprecated.Api.SpecificationRequestParameters.QueryParameters != nil {
+				q := (graphql.QueryParams)(*deprecated.Api.SpecificationRequestParameters.QueryParameters)
+				outApi.Spec.FetchRequest.Auth.AdditionalQueryParams = &q
+			}
+		}
+		out.APIDefinitions = []*graphql.APIDefinitionInput{outApi}
+
+	}
+
+	if deprecated.Events != nil && deprecated.Events.Spec != nil {
+		out.EventDefinitions = []*graphql.EventDefinitionInput{
+			{
+				Spec: &graphql.EventSpecInput{
+					Data: ptrClob(graphql.CLOB(deprecated.Events.Spec)),
+				},
+			},
+		}
+	}
+
+	if deprecated.Documentation != nil {
+		// TODO later
```

Comment:
All these `TODOs` will be done in following PRs, right?

Reply:
yes